chain/client/src/client.rs

```diff
@@ -279,7 +279,7 @@ impl Client {
             chain_config.clone(),
             snapshot_callbacks,
             async_computation_spawner.clone(),
-            validator_signer.as_ref().map(|x| x.validator_id()),
+            validator_signer.get().map(|x| x.validator_id().clone()).as_ref(),
```

Comment:
`Chain::new()` only reads validator id once, but it makes sense to pass mutable validator signer in case someone attempts to store validator id as `Chain` member or sth. Will change it.